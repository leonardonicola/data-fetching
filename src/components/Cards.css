.card {
  display: flex;
  flex-direction: column;
  background-color: #fff;
  border-radius: 20px;
  overflow: hidden;
  transition: all 0.5s ease;
  cursor: default;
}

.card__header {
  padding: 15px 0;
  height: auto;
  min-height: fit-content;
  width: 100%;
  background-color: var(--color);
  font-weight: bold;
}

.card__infos {
  padding: 15px;
  text-align: initial;
}

.card__infos span {
  font-weight: bold;
}

.card:hover {
  transform: scale(1.02);
}

input {
  margin: 20px 0;
}