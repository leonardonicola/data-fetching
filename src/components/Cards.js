import React from "react";

const Cards = ({ usersList }) => {
  return (
    <>
      {usersList &&
        usersList
          .filter((_, i) => i < 4)
          .map((user) => (
            <div key={user.id} className="card">
              <div className="card__header">
                <h4>{user.name}</h4>
              </div>
              <div className="card__infos">
              <p>
                  <span>ID: </span>
                  {user.id}
                </p>
                <p>
                  <span>Username: </span>
                  {user.username}
                </p>
                <p>
                  <span>Email: </span> 
                  {user.email}
                </p>
              </div>
            </div>
          ))}
    </>
  );
};

export default Cards;
