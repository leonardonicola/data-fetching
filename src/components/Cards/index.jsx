import "./styles.css";

const Cards = ({ title, body, id }) => {
  return (
    <>
      {!!title && (
        <div key={id} className="card">
          <div className="card__header">
            <h4 data-testid="post-title">{title}</h4>
          </div>
          <div className="card__infos">
            <p>{body}</p>
          </div>
        </div>
      )}
    </>
  );
};

export default Cards;
