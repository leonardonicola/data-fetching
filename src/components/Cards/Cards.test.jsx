import { render, screen } from "@testing-library/react";
import Cards from '.'

const mock = {
    title: 'title 1',
    body: 'body 1'
}

describe('<Cards/>', () => {
    it('should render a post with title and body', () => {
        render(<Cards {...mock}/>)
        expect(screen.getByRole('heading', {name: mock.title})).toBeInTheDocument()
        expect(screen.queryByText(mock.body)).toBeInTheDocument()
    })

    it('should match a snapshot', () => {
        const {container} = render(<Cards {...mock}/>)
        expect(container.firstChild).toMatchSnapshot()
    })
})