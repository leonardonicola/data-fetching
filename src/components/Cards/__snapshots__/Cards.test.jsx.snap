// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cards/> should match a snapshot 1`] = `
<div
  class="card"
>
  <div
    class="card__header"
  >
    <h4
      data-testid="post-title"
    >
      title 1
    </h4>
  </div>
  <div
    class="card__infos"
  >
    <p>
      body 1
    </p>
  </div>
</div>
`;
